modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultSemanticAnalysisPhase.java

```diff
@@ -1654,6 +1654,12 @@ public void visitNewObj(ENewObj userNewObjNode, SemanticScope semanticScope) {
                             "[" + canonicalTypeName + "/" + userArgumentsSize + "]"));
         }
 
+        if (semanticScope.getCondition(userNewObjNode, Read.class) == false) {
```

Comment:
This isn't backwards compatible. I've added this issue (https://github.com/elastic/elasticsearch/issues/70505) where we will need to come up with a way to remove semantics that don't make sense any longer. Please remove this for now so we don't break anyone unintentionally during upgrade to 7.10+.

Reply:
I understand.  However, it was broken before, so the only the changes is the error message.

Reply:
I still worry about customers on versions prior to 7.10 (where we last backported the recent refactorings to along with the bug). It's likely not very many, but given the complexity of upgrade and those types of issues I would prefer to avoid it if at all possible.

Reply:
Totally makes sense.